File: Azure/AdventureWorksOrder/AdventureWorksOrder.Model/Custom/SalesOrderItem.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AdventureWorksOrder.Model.Custom
{
    public class SalesOrderItem
    {
        public string AccountNumber { get; set; }

        public EntityInfo BillingAddress { get; set; }

        public EntityInfo CreditCard { get; set; }

        public EntityInfo CurrencyRate { get; set; }

        public EntityInfo Customer { get; set; }

        public DateTime SalesOrderDate { get; set; }

        public int SalesOrderId { get; set; }

        public string SalesOrderNumber { get; set; }

        public EntityInfo SalesPerson { get; set; }

        public EntityInfo SalesTerritory { get; set; }

        public EntityInfo ShipMethod { get; set; }

        public EntityInfo ShippingAddress { get; set; }
    }
}

File: Azure/AdventureWorksOrder/Web/src/model/sales-order-item.ts
import { EntityInfo } from "./entity-info";

export interface SalesOrderItem {
    accountNumber: string;
    billingAddress: EntityInfo;
    creditCard?: EntityInfo;
    currencyRate?: EntityInfo;
    customer: EntityInfo;
    salesOrderDate: string;
    salesOrderId: number;
    salesOrderNumber: string;
    salesPerson?: EntityInfo;
    salesTerritory?: EntityInfo;
    shipMethod: EntityInfo;
    shippingAddress: EntityInfo;
}